packages/fxa-settings/src/models/integrations/base-integration.ts

```diff
@@ -118,6 +118,19 @@ export abstract class Integration<
       }
     }
 
+    // TODO: Not 100% sure about this
```

Comment:
I started hitting failures in functional tests where this just wouldn't work. This logic seems very sound to me, but I am leery of any change to the service name code...

Reply:
TODO: Circle back here.

Reply:
Will circle back when looking at FXA-6836, which is also my ticket.